pkg/apis/core/validation/validation.go

```diff
@@ -6219,33 +6243,116 @@ func validatePodResizeContainerOrdering(newPod, oldPod *core.Pod, specPath *fiel
 	return allErrs
 }
 
-// dropCPUMemoryResourcesFromContainer deletes the cpu and memory resources from the container, and copies them from the old pod container resources if present.
-func dropCPUMemoryResourcesFromContainer(container *core.Container, oldPodSpecContainer *core.Container) {
-	dropCPUMemoryUpdates := func(resourceList, oldResourceList core.ResourceList) core.ResourceList {
-		if oldResourceList == nil {
-			return nil
-		}
-		var mungedResourceList core.ResourceList
-		if resourceList == nil {
-			mungedResourceList = make(core.ResourceList)
-		} else {
-			mungedResourceList = resourceList.DeepCopy()
+// isPodLevelResourcesResizeRequestSupported checks whether the pod supports resize requests
+// by verifying if the pod is running and has pod-level resources populated.
+func isPodLevelResourcesResizeRequestSupported(pod core.Pod) bool {
```

Comment:
The analogous function for InPlacePodVerticalScaling has a TODO to remove it after GA+3: https://github.com/kubernetes/kubernetes/blob/96867d8401409db5fc78dccbc43fb15706890427/pkg/apis/core/validation/validation.go#L6250-L6256

I suggest to leave a similar TODO comment here that we can remove it after GA+3